rskj-core/src/main/java/co/rsk/cli/tools/GenerateOpenRpcDoc.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2017 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package co.rsk.cli.tools;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.type.MapType;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+import com.google.common.annotations.VisibleForTesting;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Generates OpenRPC json doc file by merging a static template with several json files under <i>{workdir}/doc/rpc</i> dir
+ * <br>
+ * <br>
+ * The <b>template</b> file contains basic and static OpenRPC info.
+ * <p>The individual json files contain the different <b>methods</b>, <b>schemas</b> and
+ * <b>contentDescriptors</b> (as per OpenRPC definition) which are maintained by devs to document implemented RPC methods
+ * <br>
+ * <br>
+ * Required cli args:
+ * <ol>
+ *     <li><b>args[0]</b>: work directory where the json template and individual json files are present</li>
```

Comment:
due to the code 3 arguments are read. should we update this?

Reply:
(apologies, I was on vacation)

Good catch @VaWheel, added this field in the last minute and forgot to update the javadoc. Updated.